assets/js/modules/analytics-4/components/dashboard/ActivationBanner/index.js

```diff
@@ -72,8 +112,35 @@ export default function ActivationBanner() {
 		}
 	}, [ hasEditScope, registry, returnToSetupStep, setValues ] );
 
-	const handleSubmit = useCallback( () => {
+	useEffect( () => {
+		if ( hasSetupBannerError ) {
+			setStep( ACTIVATION_STEP_REMINDER );
+		}
+	}, [ hasSetupBannerError ] );
```

Comment:
I think this could use a comment explaining the rationale to a dev who isn't familiar with the issue this is solving, eg.:

```suggestion
	// If any errors are encountered, we change the current form step the user is
	// on to essentially "reset" the banner back to the first step. If the user 
	// tried to proceed but was offline, for instance, this would be detected and
	// return them to the first step of the banner rather than forcing an infinite
	// loading screen due to an error.
	//
	// See: https://github.com/google/site-kit-wp/issues/5928
	useEffect( () => {
		if ( hasSetupBannerError ) {
			setStep( ACTIVATION_STEP_REMINDER );
		}
	}, [ hasSetupBannerError ] );
```

Reply:
Added the comment, thank you for pointing it out and for composing it!